lib/property_table/event.ex

```diff
@@ -0,0 +1,24 @@
+defmodule PropertyTable.Event do
+  @moduledoc """
+  Struct sent to subscribers on property changes
+
+  * `:table` - the table generating this event
+  * `:property` - which property changed
+  * `:value` - the new value
+  * `:old_value` - the previous value (`nil` if this property is new)
+  * `:timestamp` - the timestamp (`System.monotonic_time/0`) when the changed happened
+  * `:previous_change` - the timestamp when the property changed to `:old_value`
+  * `:meta` - any additional metadata that was passed in the call that changed the property
+  """
+  defstruct [:table, :property, :value, :old_value, :timestamp, :previous_change, :meta]
+
+  @type t() :: %__MODULE__{
+          table: PropertyTable.table_id(),
+          property: PropertyTable.property(),
+          value: PropertyTable.value(),
+          old_value: PropertyTable.value(),
+          timestamp: integer(),
+          previous_change: integer,
```

Comment:
```suggestion
          previous_change_at: integer,
```
`previous_change` makes me think this event is given me a diff of the change. Maybe better to use a classifier like `*_at`?

Reply:
Yeah, I don't like it either and I don't like that it's not consistent with `:timestamp`.

Maybe `:timestamp` -> `:changed_at` and `:previous_change` to `:old_changed_at`? Either that or change the references to "old" to "previous"?

Reply:
What if we just made a map of the previous details?
```
%{
  previous: %{
    value: PropertyTable.value(),
    timestamp: integer()
  }
}
```

Also a tangent, is it useful to have a `inserted_at:` field with an unchanging timestamp value of the prop creation?

Reply:
I'm part way there. I haven't thought about an `inserted_at` field since I haven't needed it yet, but it might be handy. The current/previous timestamp came up with tracking metrics. 

I updated the PR to make the previous naming consistent. I didn't switch to a sub-map, but mostly since I wanted to think about it more.

One last thing. I normally wouldn't stack another PR on this one, but the `meta` field removal trimmed a bunch of code that doesn't need to be reviewed so I added it.